Apps/Gateway/src/ClientApp/src/stores/user.ts

```diff
@@ -0,0 +1,468 @@
+        }
+        // First time logging so there is no last login.
+        return undefined;
+    });
+
+    const isValidIdentityProvider = computed<boolean>(() => {
+        return oidcUserInfo.value === undefined
+            ? false
+            : oidcUserInfo.value.idp === "BCSC" ||
+                  oidcUserInfo.value.idp === undefined;
+    });
+
+    const userIsRegistered = computed<boolean>(() => {
+        return user.value === undefined
+            ? false
+            : user.value.acceptedTermsOfService;
+    });
+
+    const userIsActive = computed<boolean>(() => {
+        return user.value === undefined ? false : !user.value.closedDateTime;
+    });
+
+    const quickLinks = computed<QuickLink[] | undefined>(() => {
+        const preference =
+            user.value.preferences[UserPreferenceType.QuickLinks];
+        if (preference === undefined) {
+            return undefined;
+        }
+        return QuickLinkUtil.toQuickLinks(preference.value);
+    });
+
+    const isLoading = computed<boolean>(() => {
+        return status.value === LoadStatus.REQUESTED;
+    });
+
+    function setOidcUserInfo(userInfo: OidcUserInfo) {
+        user.value.hdid = userInfo.hdid;
+        oidcUserInfo.value = userInfo;
+        setLoadedStatus();
+    }
+
+    function setProfileUserData(userProfile: UserProfile) {
+        if (userProfile) {
+            PreferenceUtil.setDefaultValue(
+                userProfile.preferences,
+                UserPreferenceType.HideVaccineCardQuickLink,
+                "false"
+            );
+            PreferenceUtil.setDefaultValue(
+                userProfile.preferences,
+                UserPreferenceType.HideOrganDonorQuickLink,
+                "false"
+            );
+            PreferenceUtil.setDefaultValue(
+                userProfile.preferences,
+                UserPreferenceType.HideImmunizationRecordQuickLink,
+                "false"
+            );
+        }
+
+        user.value.acceptedTermsOfService = userProfile
+            ? userProfile.acceptedTermsOfService
+            : false;
+        user.value.hasTermsOfServiceUpdated = userProfile
+            ? userProfile.hasTermsOfServiceUpdated ?? false
+            : false;
+        user.value.lastLoginDateTimes = userProfile
+            ? userProfile.lastLoginDateTimes
+            : [];
+        user.value.closedDateTime = userProfile
+            ? userProfile.closedDateTime
+            : undefined;
+        user.value.preferences = userProfile ? userProfile.preferences : {};
+        user.value.hasEmail = !!userProfile?.email;
+        user.value.verifiedEmail = userProfile.isEmailVerified;
+        user.value.hasSMS = !!userProfile?.smsNumber;
+        user.value.verifiedSMS = userProfile.isSMSNumberVerified;
+        user.value.hasTourUpdated = userProfile.hasTourUpdated ?? false;
+
+        logger.verbose(`User: ${JSON.stringify(user.value)}`);
+
+        setLoadedStatus(patient.value.hdid !== undefined);
+    }
```

Comment:
```suggestion
    }
    
    function setEmailVerified() {
        user.value.verifiedEmail = true;
    }
```

This action was missing.